src/platform/pico/uart_channel.c

```diff
@@ -0,0 +1,240 @@
+#include "reactor-uc/platform/pico/uart_channel.h"
+#include "reactor-uc/platform/pico/pico.h"
+#include "reactor-uc/logging.h"
+#include "reactor-uc/serialization.h"
+
+#define UART_CHANNEL_ERR(fmt, ...) LF_ERR(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_WARN(fmt, ...) LF_WARN(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_INFO(fmt, ...) LF_INFO(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+#define UART_CHANNEL_DEBUG(fmt, ...) LF_DEBUG(NET, "UartPolledChannel: " fmt, ##__VA_ARGS__)
+
+extern Environment *_lf_environment;
+
+UartPolledChannel *uart_channel_0 = NULL;
```

Comment:
Couldn't these, and the FederateMessage be on the UartPollchannel struct instead as global variables?

Reply:
If they have to be global, lets make them static at least

Reply:
Ok, the UartPolledChannel pointers must be global to be accessible from the IRQs, but the FederateMessage does not need it I think